From 9b4943dfedfee49e250023582b09662242fefb61 Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Fri, 24 Oct 2014 02:23:44 +0200
Subject: [PATCH] net: wireless:bcmdhd: Add ENABLE_INSMOD_NO_FW_LOAD when not
 built in

And not only when the driver is built in.
Without this flag, the firmware and the NVRAM files won't be
fetched when built as a module.

Change-Id: Ib5eaeba17141a1497e65770d695b6c1f8de2e1c5
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/net/wireless/bcmdhd/Makefile    |  5 ++++-
 drivers/net/wireless/bcmdhd/dhd_linux.c | 12 +++---------
 2 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index 0a393b0..872427d 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -205,8 +205,11 @@ endif
 
 #EXTRA_LDFLAGS += --strip-debug
 
-ifeq ($(CONFIG_BCMDHD),y)
+ifneq ($(CONFIG_BCMDHD),)
   DHDCFLAGS += -DENABLE_INSMOD_NO_FW_LOAD
+endif
+
+ifeq ($(CONFIG_BCMDHD),y)
   DHDCFLAGS += -DUSE_LATE_INITCALL_SYNC
 endif
 
diff --git a/drivers/net/wireless/bcmdhd/dhd_linux.c b/drivers/net/wireless/bcmdhd/dhd_linux.c
index 118b4ec..d0155ec 100644
--- a/drivers/net/wireless/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -4699,12 +4699,9 @@ bool dhd_update_fw_nv_path(dhd_info_t *dhdinfo)
 
 	/* Update firmware and nvram path. The path may be from adapter info or module parameter
 	 * The path from adapter info is used for initialization only (as it won't change).
-	 *
-	 * The firmware_path/nvram_path module parameter may be changed by the system at run
-	 * time. When it changes we need to copy it to dhdinfo->fw_path. Also Android private
-	 * command may change dhdinfo->fw_path. As such we need to clear the path info in
-	 * module parameter after it is copied. We won't update the path until the module parameter
-	 * is changed again (first character is not '\0')
+	 * ability to change the fw path from the private cmds is disabled in
+	 * order to keep module parameter available when
+	 * ENABLE_INSMOD_NO_FW_LOAD is set.
 	 */
 
 	/* set default firmware and nvram path for built-in type driver */
@@ -4758,9 +4755,6 @@ bool dhd_update_fw_nv_path(dhd_info_t *dhdinfo)
 		       dhdinfo->nv_path[nv_len-1] = '\0';
 	}
 
-	/* clear the path in module parameter */
-	firmware_path[0] = '\0';
-
 #ifndef BCMEMBEDIMAGE
 	/* fw_path and nv_path are not mandatory for BCMEMBEDIMAGE */
 	if (dhdinfo->fw_path[0] == '\0') {
-- 
1.9.1

